Makefile

```diff
@@ -40,8 +40,8 @@ populate: create_db  ## Build and run containers
 .PHONY: test
 test: start  ## Run tests
 	if [ -z "$(name)" ]; \
-	    then FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/; \
-	    else FLASK_ENV=testing docker-compose run --rm web pytest -n 4 --dist=loadfile -v tests/ -k $(name); \
+	    then FLASK_ENV=testing docker-compose run --rm web pytest -n $$(nproc --all) --dist=loadfile -v tests/; \
```

Comment:
heads up that this is not platform independent as is (`nproc` not supported on mac out of the box), we could either put into a separate PR or revert this change? (rest of this PR looks good)

Alternatively the mac equivalent is `sysctl -n hw.logicalcpu` so one could use that if `Darwin` is the output of `uname`

Reply:
Good call, I'll update that to check.